File: LeetCode/1.Two Sum/Two Sum.java
// Problem: 1. Two Sum (Easy)
// Link: https://leetcode.com/problems/two-sum

import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();

        int idx = 0;
        for (int num : nums) {
            int tmp = target - num;
            if (map.containsKey(tmp)) {
                return new int[]{map.get(tmp), idx};
            }
            map.put(num, idx);
            idx++;
        }
        return null;
    }
}

File: LeetCode/1.Two Sum/Two Sum.py
// Problem: 1. Two Sum (Easy)
// Link: https://leetcode.com/problems/two-sum

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for index, num in enumerate(nums):
            t = target - num
            if t in map:
                return [map.get(t), index]
            map[num] = index
        